desktop/MeshService.ts

```diff
@@ -26,25 +26,27 @@ class MeshService extends NetServiceFactory {
   sendAccountMeshDataQuery = ({ accountId, offset }: { accountId: Uint8Array; offset: number }) =>
     new Promise((resolve) => {
       // @ts-ignore
-      this.service.AccountMeshDataQuery({ filter: { accountId, accountMeshDataFlags: 1 }, minLayer: { number: 0 }, maxResults: 50, offset }, (error, response) => {
-        if (error) {
-          logger.error('grpc AccountMeshDataQuery', error);
-          resolve({ totalResults: 0, data: null, error });
-        } else if (!response || !response.data) {
-          resolve({ data: [], totalResults: 0, error: null });
-        } else {
-          const { data, totalResults } = response;
-          resolve({ data, totalResults, error: null });
+      this.service.AccountMeshDataQuery(
+        { filter: { accountId: { address: accountId }, accountMeshDataFlags: 1 }, minLayer: { number: 0 }, maxResults: 50, offset },
+        (error, response) => {
+          if (error) {
+            logger.error('grpc AccountMeshDataQuery', error);
+            resolve({ totalResults: 0, data: null, error });
```

Comment:
I've already met such a pattern.
I'm sure that it's much better to reject the promise with an error and handle it as needed on the consumers` side. Probably, in this way, you don't have to check `data` for being `null` and there will be no third-state for `data` at all.

Reply:
I prefer to keep main process as free from try/catch as it possible and pass error to renderer to let user decide what to do